### PR TITLE
feat: export useWalletList for use in dapp

### PR DESCRIPTION
**Pull request by mikasackermn**
# Summary

We needed to be able to use useWalletList in the dapp

Fixes # (issue)

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: widget/embedded/src/hooks/useWalletList.ts [hunk 1]
```diff
@@ -93,6 +94,7 @@ export function useWalletList(params: Params) {
         onConnect?.(type);
       }
     } catch (e) {
+      onHandleError?.((e as any)?.message);
       setError('Error: ' + (e as any)?.message);
     }
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/hooks/useWalletList.ts:99]:
```suggestion
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
    } catch (e: any) {
      onHandleError?.(e?.message);
      setError('Error: ' + e?.message);
    }
```

File: widget/embedded/src/hooks/useWalletList.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ interface Params {
   chain?: string;
   onBeforeConnect?: (walletType: string) => void;
   onConnect?: (walletType: string) => void;
+  onHandleError?: (error?: string) => void;
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/hooks/useWalletList.ts:30]:
```suggestion
  onError?: (error?: string) => void;
```

File: widget/embedded/src/hooks/useWalletList.ts (hunk 1 above)
Comment (mikasackermn) [reply widget/embedded/src/hooks/useWalletList.ts]: fixed

File: widget/embedded/src/hooks/useWalletList.ts (hunk 2 above)
Comment (mikasackermn) [reply widget/embedded/src/hooks/useWalletList.ts]: fixed

**Review by Ikari-Shinji-re — APPROVED** [review]

**Pull request merged**